Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/connectmgr/persistance/dao/ConnectionManagerDAOBase.java

```diff
@@ -56,37 +61,43 @@
-                spf.setFeature(NhincConstants.FEATURE_DISALLOW_DOCTYPE, true);
-                spf.setFeature(NhincConstants.FEATURE_PARAMETER_ENTITIES, false);
-
-                final XMLReader xmlReader = spf.newSAXParser().getXMLReader();
-                final InputSource inputSource = new InputSource(new FileReader(file));
-                final SAXSource source = new SAXSource(xmlReader, inputSource);
-
-                final JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
-                final Unmarshaller unmarshaller = context.createUnmarshaller();
-
-                final InputSource stream = source.getInputSource();
-
-                final Reader reader = stream.getCharacterStream();
-
-                final JAXBElement<BusinessDetail> jaxbElement = unmarshaller.unmarshal(new StreamSource(reader),
-                        BusinessDetail.class);
-
+                JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
+                final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+
+                dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+                dbf.setFeature(NhincConstants.FEATURE_GENERAL_ENTITIES, false);
+                dbf.setFeature(NhincConstants.FEATURE_DISALLOW_DOCTYPE, true);
+                dbf.setFeature(NhincConstants.FEATURE_PARAMETER_ENTITIES, false);
+                dbf.setExpandEntityReferences(false);
+
+                // Convert File to Document
+                final DocumentBuilder db = dbf.newDocumentBuilder();
+                final Document doc = db.parse(file);
+                doc.getDocumentElement().normalize();
+                DOMSource source = new DOMSource(doc);
+
+                // Convert Document to StreamSource
+                TransformerFactory factory = TransformerFactory.newInstance();
+                Transformer transformer = factory.newTransformer();
+                StreamResult result = new StreamResult();
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+                result.setOutputStream(out);
+                transformer.transform(source, result);
+                ByteArrayInputStream in = new ByteArrayInputStream(out.toByteArray());
+
+                // Unmarshal
+                Unmarshaller unmarshaller = context.createUnmarshaller();
+                JAXBElement<BusinessDetail> jaxbElement = unmarshaller.unmarshal(new StreamSource(in),
+                    BusinessDetail.class);
                 resp = jaxbElement.getValue();
-
             }
-        } catch (ParserConfigurationException | SAXException | FileNotFoundException e) {
-            LOG.error("Exception in reading/parsing XDRConfiguration file: {}", e.getLocalizedMessage(), e);
+        } catch (final IOException | ParserConfigurationException | TransformerException | SAXException e) {
+            LOG.error(NhincConstants.UNABLE_LOAD_XDR_FILE, e.getLocalizedMessage(), e);
```

Comment:
We don't need to convert this exception msg into constant. Previously, sonarqube flag this since you don't have multiple catch and use the same msg.